### ISSUE TITLE
[Feature] Single-use cookie option for setCookie()

### ISSUE THREAD

**Issue by shaneAPT** [silkimen/cordova-plugin-advanced-http#472]
**Is your feature request related to a problem? Please describe.**
Make it more visible that setCookie() will persist the cookie you set across local storage (?) and it will not be removed until you explicitly call clearCookies() or removeCookies().

After getting stung with this. In a http-interceptor there is a check to see if a cookie is being passed as a header, if so setCookie() that cookie. I thought this worked fine, however logging out and logging into different users was passing incorrect cookies to my API's from whatever user I originally logged in with. This is a big no-no. Eventually I narrowed it down to setCookie persisting the cookie in local storage (?) and not removing this cookie until clearCookies() is called on logout. Even when the new user would log in and setCookie() themselves, it would not overwrite the original user cookie.

This seems to work fine now by calling clearCookies() on logout, it just took me a while to figure out that this was happening and I think it's something easy to get caught on. I could be wrong, but I couldn't find anything in the documentation giving any indication that this is the behaviour of setCookie(). My assumption was on get() the cookies would reset themselves and not persist for every subsequent call - although I do understand this may be desired.

**Describe the solution you'd like**
Add a toggle/option in setCookie() whether or not you want the given cookie to persist after it is first used in a HTTP call. If set to false, clear the cookie after the first call. This is a single-use cookie.

**Describe alternatives you've considered**
Perhaps updating the documentation to explain that setCookie() cookies will not be removed until removeCookies() or clearCookies() is called.

**Comment by silkimen** [comment]
Hi @shaneAPT, thank you for your feedback!

Yes, you are right about the behaviour you were experiencing. The `setCookie()` function is meant to **ADD** cookies into the cookie storage. I guess naming it `addCookie` would've been better, but that'd be a breaking change now. 😣

You need to remove those cookies manually by either using `clearCookies()` (which you already found out) or using `removeCookies()` which will allow you to remove only the cookies associated with a given URL.

So basically, this ticket consists of two parts:
1. Documentation should be updated accordingly to explain current behaviour. Would you please create a PR?
2. You are requesting a new feature. I can't look into this now, but maybe in the future. If someone (maybe you?) is willing to draft a PR, they are welcome! 😊❤️

**Comment by shaneAPT** [comment]
Thanks for getting back to me @silkimen. I created a PR there to tweak the README. 

After giving it more thought, perhaps the ability to add cookies only temporarily is better suited for an update to the ToughCookie plugin? As I see putCookie is called and that gives no option to add per session or temporarily.
